src/rpp/rpp/operators/element_at.hpp

```diff
@@ -0,0 +1,99 @@
+//                  ReactivePlusPlus library
+//
+//          Copyright Aleksey Loginov 2023 - present.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          https://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/victimsnino/ReactivePlusPlus
+//
+
+#pragma once
+
+#include <rpp/operators/fwd.hpp>
+
+#include <rpp/defs.hpp>
+#include <rpp/operators/details/strategy.hpp>
+
+#include <cstddef>
+
+namespace rpp::operators::details
+{
+    template<rpp::constraint::observer TObserver>
+    struct element_at_observer_strategy
+    {
+        using preferred_disposable_strategy = rpp::details::observers::none_disposable_strategy;
+
+        RPP_NO_UNIQUE_ADDRESS TObserver observer;
+        size_t                          index{};
+        mutable size_t                  current{};
+
+        template<typename T>
+        void on_next(T&& v) const
+        {
+            if (current++ == index)
+            {
+                observer.on_next(std::forward<T>(v));
```

Comment:
i think, we should call on_completed after on_next?

Reply:
indeed, done